Share with -> # crashes Delta Chat

<!--
This is a bug report tracker. New features are discussed in the forum: https://support.delta.chat

Please fill out as much of this form as you can (leaving out stuff that is not applicable is ok).
-->

- Android version: 7.1.2
- Device: MotoG3
- Delta Chat version: 1.3.0 (dev version)
- Actual behavior:
  In the "Direct share" section there appears a chat that has a white `#` on black background coming form Delta Chat. When I click it, DC crashes.
- Steps to reproduce the problem:
  - Click in "Share" in another app
- Logs:
```
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.b44t.messenger.beta/org.thoughtcrime.securesms.ConversationActivity}: java.lang.IllegalStateException: can't display a conversation for no chat.
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2684)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6186)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
     Caused by: java.lang.IllegalStateException: can't display a conversation for no chat.
        at org.thoughtcrime.securesms.ConversationActivity.initializeResources(ConversationActivity.java:835)
        at org.thoughtcrime.securesms.ConversationActivity.onCreate(ConversationActivity.java:225)
        at org.thoughtcrime.securesms.PassphraseRequiredActionBarActivity.onCreate(PassphraseRequiredActionBarActivity.java:34)
        at android.app.Activity.performCreate(Activity.java:6684)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2637)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751) 
        at android.app.ActivityThread.-wrap12(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6186) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779) 
```

I have no such chat in my chat list, no wonder DC crashes. But why does this icon appear?
<!--
Debug logs can be copied from within the Delta Chat app with
Settings menu -> Advanced -> Debug log

Alternatively from the Android system log:
`adb logcat -v time -s DeltaChat`

Logs may contain private data 
which shall be removed or anonymised prior to posting.
-->

--- a/src/org/thoughtcrime/securesms/service/DirectShareService.java
+++ b/src/org/thoughtcrime/securesms/service/DirectShareService.java
@@ -48,6 +48,9 @@ public class DirectShareService extends ChooserTargetService {
     }
     for (int i = 0; i <= max; i++) {
       DcChat chat = chatlist.getChat(i);
+      if (!chat.canSend()) {
+        continue;
+      }
 
       Bundle bundle = new Bundle();
       bundle.putInt(ShareActivity.EXTRA_CHAT_ID, chat.getId());

--- a/src/org/thoughtcrime/securesms/service/DirectShareService.java
+++ b/src/org/thoughtcrime/securesms/service/DirectShareService.java
@@ -41,7 +41,11 @@ public class DirectShareService extends ChooserTargetService {
     ComponentName       componentName  = new ComponentName(this, ShareActivity.class);
     ApplicationDcContext dcContext = DcHelper.getContext(this);
 
-    DcChatlist chatlist = dcContext.getChatlist(DcContext.DC_GCL_ADD_ALLDONE_HINT, null, 0);
+    DcChatlist chatlist = dcContext.getChatlist(
+            DcContext.DC_GCL_ADD_ALLDONE_HINT | DcContext.DC_GCL_FOR_FORWARDING | DcContext.DC_GCL_NO_SPECIALS,
+            null,
+            0
+    );
     int max = 4;
     if (chatlist.getCnt() < max) {
       max = chatlist.getCnt();

